### PR TITLE
Update docs

### PR DESCRIPTION
**Pull request by jGaboardi**
This PR addresses #336 (from pysal/submodule_template#23)

### CODE REVIEW

File: docsrc/conf.py [hunk 1]
```diff
@@ -246,17 +237,21 @@
...
 
+
 def setup(app):
     app.add_stylesheet("pysal-styles.css")
 
-# Example configuration for intersphinx: refer to the Python standard library.
-intersphinx_mapping = {'https://docs.python.org/3.6/': None}
 
+# Example configuration for intersphinx: refer to the Python standard library.
+intersphinx_mapping = {"https://docs.python.org/3.6/": None}
```
Comment (knaaptime) [inline docsrc/conf.py]: 'numpy': ('https://docs.scipy.org/doc/numpy', None),
    'pandas': ('https://pandas.pydata.org/pandas-docs/stable/', None),
    'geopandas': ('https://geopandas.readthedocs.io/en/latest/', None),
    'sklearn': ('https://scikit-learn.org/stable/', None),
    'giddy': ('https://giddy.readthedocs.io/en/latest/', None),
    'libpysal': ('https://pysal.org/libpysal/', None),
    'esda': ('https://esda.readthedocs.io/en/latest/', None),
    'region': ('https://region.readthedocs.io/en/latest/', None)
}

**Comment by jGaboardi** [comment]
@weikang9009 @knaaptime 
What's going on [here](https://travis-ci.org/pysal/spaghetti/jobs/621746862#L579)? Any ideas?

```
The program 'nosetests' is currently not installed.
To run 'nosetests' please ask your administrator to install the package 'python-nose'.
```

File: docsrc/conf.py (hunk 1 above)
Comment (jGaboardi) [reply docsrc/conf.py]: AH, yes! Will update that.

**Comment by jGaboardi** [comment]
> Looks great! I think you can safely remove `doc/` directory since the docs website is no longer hosted with readthedocs.

@weikang9009 , so I can completely remove the `doc/` directory from both the `remote` and my `local` copy?

**Comment by weikang9009** [comment]
> > Looks great! I think you can safely remove `doc/` directory since the docs website is no longer hosted with readthedocs.
> 
> @weikang9009 , so I can completely remove the `doc/` directory from both the `remote` and my `local` copy?

That's correct. Since you are hosting with github page, all you need is the docsrc/ and doc/

**Comment by knaaptime** [comment]
i think you need *docs* (and docsrc) but not doc

**Comment by jGaboardi** [comment]
>  docsrc/ and doc/

Do you mean docsrc/ and doc****s****/

**Comment by weikang9009** [comment]
yes. just corrected it.

File: docsrc/conf.py (hunk 1 above)
Comment (knaaptime) [reply docsrc/conf.py]: maybe scipy too? not sure what you use in spaghetti

Comment (jGaboardi) [reply docsrc/conf.py]: doh, yeah. 

Also, should I have kept `'https://docs.python.org/3.6/': None` or is that OK to just get rid of?

Comment (knaaptime) [reply docsrc/conf.py]: sorry, i meant those as addition not replacement. If you keep `'https://docs.python.org/3.6/': None` you get formatting for basic python primitives like str and int and dict

**Comment by jGaboardi** [comment]
> @weikang9009 @knaaptime
What's going on here? Any ideas?
>```The program 'nosetests' is currently not installed.```
> ```To run 'nosetests' please ask your administrator to install the package 'python-nose'.```

Also getting
```
The program 'twine' is currently not installed.
To run 'twine' please ask your administrator to install the package 'twine'
```
in the basic [`tarball` builds](https://travis-ci.org/pysal/spaghetti/jobs/621760391). Seems like I may have screwed something up, can't find how anything (substantive) has changed. Maybe it's just a temporary Travis issue?

**Comment by knaaptime** [comment]
that looks like a different travis config to me? on line 383 its doing `pip install -e . --no-deps --force-reinstall`

insteead of [using conda](https://github.com/pysal/spaghetti/blob/b9efc62401e5d2cd3bf0a1d36b858e8a99d31e52/.travis.yml#L54) to install from the `requirements_*` files

**Comment by knaaptime** [comment]
oh wait, nvm looks like [that line](https://github.com/pysal/spaghetti/blob/b9efc62401e5d2cd3bf0a1d36b858e8a99d31e52/.travis.yml#L59) comes shortly after.

i dunno...

**Comment by jGaboardi** [comment]
Yeah, seems like something is going on in Travis ATM... Will revisit this later.

**Comment by jGaboardi** [comment]
It's strange that only `spaghetti` is failing right now though, but not any other submodules.

**Comment by jGaboardi** [comment]
`- sphinx_bootstrap_theme[version='>=0.7.0']` seems to be causing [the issue](https://github.com/pysal/spaghetti/blob/master/.travis.yml).

```
301 Solving environment: failed
302
303 PackagesNotFoundError: The following packages are not available from current channels:
304
305  - sphinx_bootstrap_theme[version='>=0.7.0']
306
307 Current channels:
308
309  - https://conda.anaconda.org/conda-forge/linux-64
310  - https://conda.anaconda.org/conda-forge/noarch
311  - https://repo.anaconda.com/pkgs/main/linux-64
312  - https://repo.anaconda.com/pkgs/main/noarch
313  - https://repo.anaconda.com/pkgs/r/linux-64
314  - https://repo.anaconda.com/pkgs/r/noarch
315
316 To search for alternate channels that may provide the conda package you're
317 looking for, navigate to
318
319    https://anaconda.org
320
321 and use the search bar at the top of the page.
322
323
324 Could not find conda environment: TEST
```

**Comment by knaaptime** [comment]
ah, thats my bad. the newest sphinx boostrap theme isnt on conda yet. you can install it from pip in the travis build so you can keep the environment.yml untouched for when its actually conda released

**Comment by jGaboardi** [comment]
I forgot to actually ***rebuild*** the docs... Getting a build error.

```
# Sphinx version: 1.8.5
# Python version: 3.6.7 (CPython)
# Docutils version: 0.15.2 release
# Jinja2 version: 2.10.3
# Last messages:
#   reading sources... [ 72%] generated/spaghetti.Network.savenetwork
#   
#   reading sources... [ 75%] generated/spaghetti.Network.simulate_observations
#   
#   reading sources... [ 78%] generated/spaghetti.Network.snapobservations
#   
#   reading sources... [ 81%] generated/spaghetti.Network.split_arcs
#   
#   reading sources... [ 84%] generated/spaghetti.PointPattern
#   
# Loaded extensions:
#   sphinx.ext.mathjax (1.8.5) from /Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/ext/mathjax.py
#   alabaster (0.7.12) from /Users/user/miniconda3/envs/env/lib/python3.6/site-packages/alabaster/__init__.py
#   sphinx.ext.autodoc (1.8.5) from /Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/ext/autodoc/__init__.py
#   sphinx.ext.autosummary (1.8.5) from /Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/ext/autosummary/__init__.py
#   sphinx.ext.viewcode (1.8.5) from /Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/ext/viewcode.py
#   sphinxcontrib.bibtex (unknown version) from /Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinxcontrib/bibtex/__init__.py
#   sphinx.ext.doctest (1.8.5) from /Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/ext/doctest.py
#   sphinx.ext.intersphinx (1.8.5) from /Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/ext/intersphinx.py
#   numpydoc (0.8.0) from /Users/user/miniconda3/envs/env/lib/python3.6/site-packages/numpydoc/__init__.py
#   matplotlib.sphinxext.plot_directive (3.1.2) from /Users/user/miniconda3/envs/env/lib/python3.6/site-packages/matplotlib/sphinxext/plot_directive.py
Traceback (most recent call last):
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/cmd/build.py", line 304, in build_main
    app.build(args.force_all, filenames)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/application.py", line 341, in build
    self.builder.build_update()
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 347, in build_update
    len(to_build))
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 360, in build
    updated_docnames = set(self.read())
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 468, in read
    self._read_serial(docnames)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 490, in _read_serial
    self.read_doc(docname)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/builders/__init__.py", line 534, in read_doc
    doctree = read_doc(self.app, self.env, self.env.doc2path(docname))
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/io.py", line 318, in read_doc
    pub.publish()
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/core.py", line 217, in publish
    self.settings)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/readers/__init__.py", line 71, in read
    self.parse()
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/readers/__init__.py", line 77, in parse
    self.parser.parse(self.input, document)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/parsers.py", line 88, in parse
    self.statemachine.run(inputstring, document, inliner=self.inliner)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 171, in run
    input_source=document['source'])
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 2771, in underline
    self.section(title, source, style, lineno - 1, messages)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 327, in section
    self.new_subsection(title, lineno, messages)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 395, in new_subsection
    node=section_node, match_titles=True)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 282, in nested_parse
    node=node, match_titles=match_titles)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 2346, in explicit_markup
    self.explicit_list(blank_finish)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 2376, in explicit_list
    match_titles=self.state_machine.match_titles)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 319, in nested_list_parse
    node=node, match_titles=match_titles)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 196, in run
    results = StateMachineWS.run(self, input_lines, input_offset)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/statemachine.py", line 239, in run
    context, state, transitions)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/statemachine.py", line 460, in check_line
    return method(match, context, next_state)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 2649, in explicit_markup
    nodelist, blank_finish = self.explicit_construct(match)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 2356, in explicit_construct
    return method(self, expmatch)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 2099, in directive
    directive_class, match, type_name, option_presets)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/parsers/rst/states.py", line 2148, in run_directive
    result = directive_instance.run()
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/ext/autodoc/directive.py", line 121, in run
    documenter_options = process_documenter_options(doccls, self.config, self.options)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/ext/autodoc/directive.py", line 73, in process_documenter_options
    return Options(assemble_option_dict(options.items(), documenter.option_spec))
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/docutils/utils/__init__.py", line 329, in assemble_option_dict
    options[name] = convertor(value)
  File "/Users/user/miniconda3/envs/env/lib/python3.6/site-packages/sphinx/ext/autodoc/__init__.py", line 82, in members_option
    return [x.strip() for x in arg.split(',')]
AttributeError: 'bool' object has no attribute 'split'
```

**Comment by weikang9009** [comment]
Try removing conda install requirements_docs.txt (https://github.com/pysal/spaghetti/blob/master/.travis.yml#L54) from the travis file. We do not need the docs packages for travis testing

**Comment by jGaboardi** [comment]
> Try removing conda install requirements_docs.txt from the travis file. We do not need the docs packages for travis testing.

This does not seem to be the issue as the current (`spaghetti`) build is passing in Travis. The [error above](https://github.com/pysal/spaghetti/pull/337#issuecomment-562747738) happens when attempting `make html` locally.

**Comment by weikang9009** [comment]
I see. Maybe try updating your local sphinx? You are using a pretty old version.

Conda install the docs dependencies for the travis testing seems to be redundant - and could slow down the testing process

**Comment by jGaboardi** [comment]
> I see. Maybe try updating your local sphinx? You are using a pretty old version.

Perfect solution!

File: .travis.yml [hunk 2]
```diff
@@ -52,6 +52,7 @@ before_install:
     conda config --set channel_priority strict
     conda config --set safety_checks disabled
     conda create --name TEST python=$PY --file requirements.txt --file requirements_dev.txt --file requirements_docs.txt --file requirements_tests.txt
```
Comment (jGaboardi) [inline .travis.yml]: @weikang9009  So I should be able to completely remove both of the follow?:
 * `requirements_docs.txt` from Line 54 and;
 * `pip install 'sphinx_bootstrap_theme>=0.7.0' --force-reinstall` from Line 55

**Pull request merged**